containers.go

```diff
@@ -309,10 +309,10 @@ func (r *containerStore) Delete(id string) error {
 		return ErrContainerUnknown
 	}
 	id = container.ID
-	newContainers := []*Container{}
-	for _, candidate := range r.containers {
-		if candidate.ID != id {
-			newContainers = append(newContainers, candidate)
+	toDeleteIndex := -1
+	for i, candidate := range r.containers {
```

Comment:
I guess, question for @nalind , I suppose `r.containers` and `r.layers` can't contain duplicates, or do they? I'll edit this patch if they do cause here we're just filtering out 1 element.